kaldo/observables/harmonic_with_q.py

```diff
@@ -302,24 +298,26 @@ def calculate_eigensystem_unfolded(self, only_eigenvals=False):
         fc_s = fc_s.reshape((n_unit_cell, 3, supercell[0], supercell[1], supercell[2], n_unit_cell, 3))
         supercell_positions = self.second.supercell_positions
         supercell_norms = 1 / 2 * np.linalg.norm(supercell_positions, axis=1) ** 2
-        dyn_s = np.zeros((n_unit_cell, 3, n_unit_cell, 3), dtype=np.complex)
+        dyn_s = np.zeros((n_unit_cell, 3, n_unit_cell, 3), dtype=complex)
         supercell_replicas = self.second.supercell_replicas
         for ind in range(supercell_replicas.shape[0]):
             supercell_replica = supercell_replicas[ind]
-            replica_position = np.tensordot(supercell_replica, cell, (-1, 0))
+            replica_position = np.tensordot(supercell_replica, cell.T, (-1, 0))
             distance = replica_position[None, None, :] + (atoms.positions[:, None, :] - atoms.positions[None, :, :])
             projection = (contract('la,ija->ijl', supercell_positions, distance) - supercell_norms[None, None, :])
             mask = (projection <= 1e-6).all(axis=-1)
             neq = (np.abs(projection) <= 1e-6).sum(axis=-1)
             weight = 1.0 / neq
             coefficient = weight * mask
             if coefficient.any():
+                image = supercell_replica%supercell
```

Comment:
Can you add a comment on what is `image` or use a more meaningful name, please?
This way we know next time. :)